client/src/views/files/FolderSelectionModal.vue

```diff
@@ -0,0 +1,229 @@
+<!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
+
+<template>
+  <ms-modal
+    :title="title"
+    :subtitle="subtitle"
+    :close-button="{ visible: true }"
+    :cancel-button="{
+      label: $t('TextInputModal.cancel'),
+      disabled: false,
+      onClick: cancel,
+    }"
+    :confirm-button="{
+      label: $t('TextInputModal.ok'),
+      disabled: selectedPath === startingPath,
+      onClick: confirm,
+    }"
+  >
+    <header-breadcrumbs
+      :path-nodes="headerPath"
+      @change="onPathChange"
+    />
+    <div>
+      <ion-list>
+        <ion-list-header
+          class="folder-list-header"
+          lines="full"
+        >
+          <ion-label class="folder-list-header__label label-name">
+            {{ $t('FoldersPage.listDisplayTitles.name') }}
+          </ion-label>
+        </ion-list-header>
+        <div class="folders-container">
+          <ion-item
+            class="file-list-item"
+            v-for="entry in currentEntries"
+            :key="entry.id"
+            :disabled="entry.isFile()"
+            @click="enterFolder(entry)"
+          >
+            <div class="file-name">
+              <div class="file-name__icons">
+                <ion-icon
+                  class="main-icon"
+                  :icon="entry.isFile() ? document : folder"
+                  size="default"
+                />
+              </div>
+              <ion-label class="file-name__label cell">
+                {{ entry.name }}
+              </ion-label>
+            </div>
+          </ion-item>
+        </div>
+      </ion-list>
+    </div>
+  </ms-modal>
+</template>
+
+<script lang="ts">
+import FolderSelectionModal from '@/views/files/FolderSelectionModal.vue';
+
+export interface FolderSelectionOptions {
+  title: string;
+  subtitle?: string;
+  startingPath: FsPath;
+}
+
+export async function selectFolder(options: FolderSelectionOptions): Promise<FsPath | null> {
+  const modal = await modalController.create({
+    component: FolderSelectionModal,
```

Comment:
I guess that my previous comment is related to this function, as discussed some weeks ago, really not fan of having this functions in the component / view directly, it should be in a separated TS file.
Easier to read, easier to understand and avoid weird imports as above.

Reply:
Yeah I've put it here for now but we should move all those input functions to one file eventually. I created an issue #5847